pyomo/gdp/plugins/fix_disjuncts.py

```diff
@@ -38,20 +43,33 @@ def __init__(self, **kwargs):
     ))
 
     def _apply_to(self, model, **kwds):
-        """Fix all disjuncts in the given model and reclassify them to Blocks."""
+        """Fix all disjuncts in the given model and reclassify them to 
+        Blocks."""
         config = self.config = self.CONFIG(kwds.pop('options', {}))
         config.set_value(kwds)
 
         self._transformContainer(model)
 
         # Reclassify all disjuncts
-        for disjunct_object in model.component_objects(Disjunct, descend_into=(Block, Disjunct)):
-            disjunct_object.parent_block().reclassify_component_type(disjunct_object, Block)
+        for disjunct_object in model.component_objects(Disjunct,
+                                                       descend_into=(Block,
+                                                                     Disjunct)):
+            disjunct_object.parent_block().reclassify_component_type(
+                disjunct_object, Block)
+
+        # Transform any remaining logical stuff
+        TransformationFactory('core.logical_to_linear').apply_to(model)
```

Comment:
Is it worth having an option to control whether or not this happens?

Reply:
I thought about this, but I cannot think of any application where you would like to have lingering logical constraints after fixing disjunct. Do you have any examples in mind?

Reply:
I'm inclined to agree with @bernalde. I think you may as well head all the way to algebraic-land (assuming the only stuff that needed transformation on the model *was* GDP stuff) because there's not really any way to go back to something logical after calling this.